tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs

```diff
@@ -207,5 +209,30 @@ internal static string GetLogDirectory()
 
             return logDirectory;
         }
+
+        internal static void CleanLogFiles(string logsDirectory)
+        {
+            DirectoryInfo logsDirInfo = new DirectoryInfo(logsDirectory);
```

Comment:
I recommend using the static `Directory` and `File` classes instead of creating the `DirectoryInfo` instance and the _two_ potentially unbounded `FileInfo` arrays. `EnumerateFiles()` let's you iterate over the filenames without creating the array in memory. Lastly, you don't need to sort the files, you just need to find the ones older than 31 days.

Something like (consider this pseudo-code, I didn't try to compile it):
```csharp
var date = DateTime.Now.AddDays(-31);

foreach (var filename in Directory.EnumerateFiles(logsDirectory))
{
    if (File.GetLastWriteTime(filename) < date)
    {
        try
        {
            File.Delete(filename);
        }
        catch
        {
            // ignore
        }
    }
}
```

Reply:
Sounds good, I was trying to sort then to avoid going over all the files in case they had a lot but I guess the sorting would be doing that extra work.

Will give these a try!